engine/src/test/java/org/hibernate/search/test/util/impl/ClasspathResourceAsFile.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.test.util.impl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.junit.rules.ExternalResource;
+
+
+/**
+ * Provides access to a file that is a copy of a given classpath resource.
+ *
+ * <p>Useful to test code that expects actual files on the filesystem using
+ * resources from a different jar (which are zipped and are not actual files).
+ *
+ * @author Yoann Rodiere
+ */
+public class ClasspathResourceAsFile extends ExternalResource {
+
+	private File parentDirectory;
+
+	private final URL url;
+
+	private File file;
+	private boolean fileIsTemporary;
```

Comment:
Maybe `hasCreatedTempFile` would be better? I thought it states if the file should be temporary or permanent.